snooty/parser.py

```diff
@@ -616,42 +618,66 @@ def handle_directive(
                     self.validate_list_table(bullets, expected_num_columns)
 
         elif name == "openapi":
-            if argument_text is None:
-                self.diagnostics.append(ExpectedPathArg(name, line))
-                return doc
+            # Parsing should be done by the OpenAPI renderer on the frontend by default
+            parse_method = options.get("parse-method", "filepath")
 
-            openapi_fileid, filepath = util.reroot_path(
-                FileId(argument_text), self.docpath, self.project_config.source_path
-            )
+            if parse_method == "url":
```

Comment:
I think determining whether the argument is a URL or filepath is straightforward enough that it can be handled on the parser. IMO, not requiring an option to switch between the two is a preferable UX experience worth implementing.

Reply:
Done!